### DISCUSSION TITLE
To those who want more all kinds of UI customization, you can use Maui blazor.

### DISCUSSION THREAD

**Discussion by nextcodelab** [dotnet/maui#5274] [Ideas]
Some controls in maui native is limited like.. label span, you cannot customize the look of spans like colorize the underline not text.
Maui blazor you can customize all kinds of styles in css and html...

MAUI Blazor is the most complete controls and styles than Flutter... because of html and css, css is the most complete styling engine.

You could also mix native view and web UI, Example Shell/MenuDrawer/Tabbed/Dialog is the main native UI Root, then Content is in blazor...  so designing a custom list Content is unlimited.

**Comment by Laftek** [comment]
amen! ... if its going to work properly 😄

**Reply by nextcodelab** [reply]
C# to Css styler.
[https://github.com/coderbasket/StylerCode.Css](https://github.com/coderbasket/StylerCode.Css)

**Comment by nogginbox** [comment]
You can use CSS in a MAUI XAML App.

[Maui docs on styling with CSS](https://docs.microsoft.com/en-us/dotnet/maui/user-interface/styles/css).

**Reply by matt-goldman** [reply]
I think it's important to note that this is a CSS style file to customise styles that are supported by .NET MAUI (and exposed in XAML). You can't use full web-standards CSS.

**Reply by coderbasket** [reply]
Why this style need to be available in xaml? xaml styles setter for me I think its enough for non html.
      I use blazor for content like book reader because in support click event and unlimited styling for text, but DRAWER_MENU USES NATIVE UI.

this is example below the text decoration underline color / text background color is available  in html/css but not in native UI even in WinUI3.


![Screenshot 2022-03-09 201331](https://user-images.githubusercontent.com/97447955/159154703-704bc772-116d-4995-9603-53d114177439.png)

**Comment by coderbasket** [comment]
There is a library here... with enums. css styling in c# and some cal().
each css enums has documentation and example along with various posible style values.
[https://github.com/coderbasket/StylerCode.Css](https://github.com/coderbasket/StylerCode.Css)

![alt text](https://raw.githubusercontent.com/coderbasket/StylerCode.Css/main/20220319_191849.jpg)
